src/reactor.c

```diff
@@ -30,7 +30,6 @@ void Reactor_validate(Reactor *self) {
     if (trigger->type == TRIG_INPUT || trigger->type == TRIG_OUTPUT) {
       Port *port = (Port *)trigger;
       validate(port->effects.num_registered == port->effects.size);
-      validate(port->sources.num_registered == port->sources.size);
```

Comment:
Why remove this check? It is a code-gen bug if we say that a port has N sources but then we do not register N